jetty-server/src/main/java/org/eclipse/jetty/server/SymlinkAllowedResourceAliasChecker.java

```diff
@@ -34,33 +33,42 @@
      */
     public SymlinkAllowedResourceAliasChecker(ContextHandler contextHandler)
     {
-        super(contextHandler, false);
+        super(contextHandler);
     }
 
     @Override
-    public boolean check(String uri, Resource resource)
+    public boolean check(String pathInContext, Resource resource)
     {
-        try
-        {
-            // Check the resource is allowed to be accessed.
-            if (!super.check(uri, resource))
-                return false;
+        // The existence check resolves the symlinks.
+        if (!resource.exists())
+            return false;
+
+        Path path = getPath(resource);
+        if (path == null)
+            return false;
 
-            // Approve if path is a symbolic link.
-            Path resourcePath = resource.getFile().toPath();
-            if (Files.isSymbolicLink(resourcePath))
-                return true;
+        String[] segments = pathInContext.split("/");
+        Path fromBase = _base;
 
-            // Approve if path has symlink in under its resource base.
-            if (super.hasSymbolicLink(getBasePath(), resourcePath))
-                return true;
+        try
+        {
+            for (String segment : segments)
+            {
+                fromBase = fromBase.resolve(segment);
+                if (!Files.exists(fromBase))
+                    return false;
+                if (Files.isSymbolicLink(fromBase))
+                    return true;
+                if (!isAllowed(fromBase))
+                    return false;
```

Comment:
We are walking down from the top looking for a symbolic link, which we will allow.  But anything about that must be allowed.    This is to protect against the "wEb-InF/someSymLinkNotForPublicConsumption.xml", which without this check would be allowed just because it is a symlink.  With this test, it is not allowed because the first segment "wEb-InF" is a protected resource.